### PR TITLE
Auto Populate Some Beancounters for Enemies

### PR DESCRIPTION
**Pull request by coreyjkelly**
Here's the basic idea. This populates counters for:

- Spellcasting
- Innate Spellcasting
- Legendary Actions
- Legendary Resistances
- Any Actions/Traits with the "Recharge" keyword

Adding Legendary Actions here raises the question: is there a way to have this counter reset automatically each round? Starts to overlap with tags a bit at that point. Is that worth the effort?

If the general design is okay, I'll clean it up and add tests. As usual, criticism on design patterns and best practices is appreciated.

### CODE REVIEW

**Comment by coreyjkelly** [comment]
I think this works reasonably well for everything in the OGL/SRD. I can only think of a few edge cases where it might not. Happy to tweak things as users inevitably find more edge cases. 

Not sure why an `AccountClient` test is failing - is that an issue on my end?

Also, is there any note markdown that can be used to prevent spell formatting from catching the _Resistance_ part of _Legendary Resistance_?

**Comment by cynicaloptimist** [comment]
I broke that test the other day, haven't fixed yet, sorry about that. Will
review this PR soon!

Evan Bailey
http://www.improved-initiative.com
Pledge on Patreon <https://www.patreon.com/improvedinitiative> | Contribute
on Github <https://github.com/cynicaloptimist/improved-initiative>

File: client/Combatant/Combatant.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class Combatant {
   private processCombatantState(savedCombatant: CombatantState) {
     this.IndexLabel = savedCombatant.IndexLabel || 0;
     this.CurrentHP(savedCombatant.CurrentHP);
-    this.CurrentNotes(savedCombatant.CurrentNotes || "");
+    this.CurrentNotes(savedCombatant.CurrentNotes || this.AutoPopulatedNotes());
```
Comment (cynicaloptimist) [inline client/Combatant/Combatant.ts]: I think this call probably belongs in `AddCombatantFromStatBlock()`, assigned to `initialState.CurrentNotes`. My reasoning is that this should only ever auto-populate when adding from a StatBlock. 

An interesting side effect of putting the call here is that very old savedCombatants (from before CurrentNotes) would receive AutoPopulatedNotes(), but newer ones would not.

File: client/Combatant/Combatant.ts [hunk 2]
```diff
@@ -309,4 +309,70 @@ export class Combatant {
       .filter(c => c.IsPlayerCharacter() === this.IsPlayerCharacter())
       .sort((a, b) => a.Initiative() - b.Initiative())[0];
   }
+
+  private AutoPopulatedNotes: () => string = () => {
+    let notes = "";
```
Comment (cynicaloptimist) [inline client/Combatant/Combatant.ts]: I think it would be a little cleaner if you started notes as a `string []`, pushed members into it as you go, and then call `.join("\n\n");` at the end. String concatenation is a little hazardous IMO.

**Comment by cynicaloptimist** [comment]
Dunno about adding that per-round functionality to counters. IMO, these counters are useful to track things that last for more than a single encounter (hence their presence in persistent notes). I'll keep an eye out for anyone asking for the functionality.

Unfortunately there's no way to "opt out" of keyword/spell lookups, but that might be a useful feature.

I re-ran CI, tests look good.

**Comment by coreyjkelly** [comment]
Agreed on the "per round" stuff. I thought of it while adding the Legendary Actions, since those counters will currently require the DM to remember to reset them every round. The other use case I thought of was tracking Reaction usage for larger parties (I know I lose track). This doesn't really feel like the best way to tackle that, though.
